server/db/db.js

```diff
@@ -1176,10 +1181,10 @@ async function getMostRecentSensorsVitalWithLocationid(locationid, pgClient) {
       return null
     }
 
-    const allLocations = await getLocations(pgClient)
-    return createSensorsVitalFromRow(results.rows[0], allLocations)
+    return createSensorsVitalFromRow(results.rows[0], [location])
```

Comment:
Very nice change here. Removes an unnecessary (and probably slow) query to get all locations; great catch.

Reply:
Thanks!